libnmstate/varlink/io.nmstate.varlink

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+# Interface definition
+interface io.nmstate
+
+# Method definition
+method Show(include_status_data: ?bool) -> (state: ?object)
+
+method Apply(state: object, verify_change: ?bool, commit: ?bool, rollback_timeout: ?int) -> (response: ?object)
+
+method Commit(checkpoint: ?string) -> (response: ?object)
+
+method Rollback(checkpoint: ?string) -> (response: ?object)
+
+# Error definition
+error NmstateValueError (reason: object)
```

Comment:
This `object` is for `Foreign untyped/raw object`.

I think it's reasonable to assume all errors only contain string for reason.